tensorboard/webapp/metrics/effects/index.ts

```diff
@@ -261,6 +263,23 @@ export class MetricsEffects implements OnInitEffects {
     })
   );
 
+  private readonly addOrRemovePin$ = this.actions$.pipe(
+    ofType(actions.cardPinStateToggled),
+    withLatestFrom(this.getVisibleCardFetchInfos()),
+    map(([{cardId, canCreateNewPins, wasPinned}, fetchInfos]) => {
```

Comment:
Can we protect this with a feature flag until you have the entire feature built and tested end-to-end?

I'm worried you will build out the rest of the feature and test it and find some reason to make changes to the storage layer. However, people will already have started writing their pins to local storage (even if they don't see the rest of this feature) and perhaps what they store will be incompatible with the final version.

Reply:
SGTM! Added a new [feature flag](https://github.com/tensorflow/tensorboard/pull/6819/files#diff-6714ebaef67a0cecfd99af712ed59348181b22598ef6ceef7f0f3953feeb11ecR54)